File: back/src/main/java/csh/back/domain/vote/user/entity/VoteUser.java
package csh.back.domain.vote.user.entity;

import csh.back.domain.trip.member.entity.TripMember;
import csh.back.domain.trip.timeline.entity.TimeLine;
import csh.back.domain.vote.item.entity.VoteItem;
import csh.back.domain.vote.vote.entity.Vote;
import csh.back.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "trip_vote_users",
        uniqueConstraints = @UniqueConstraint(
                name = "uk_vote_user",
                columnNames = {"vote_id", "trip_member_id"}  // vote_id 기준
        )
)

@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class VoteUser extends BaseEntity {

    //FK
    //Join Vote Table
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vote_id",  nullable = false)
    private Vote vote;

    //FK
    //Join VoteItem Table
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "vote_item_id",  nullable = false)
    private VoteItem voteItem;

    //FK
    //Join TripMember Table
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trip_member_id",  nullable = false)
    private TripMember tripMember;

    @Column(nullable = false)
    private Integer updateCount;

    //생성자
    //빌드 사용
    @Builder
    private VoteUser(Vote vote, VoteItem voteItem, TripMember tripMember, Integer updateCount) {
        this.vote = vote;
        this.voteItem = voteItem;
        this.tripMember = tripMember;
        this.updateCount = updateCount;
    }

    public void updateVoteItemAndincreaseUpdateCount(VoteItem voteItem) {
        this.voteItem = voteItem;
        this.updateCount++;
    }
}

File: back/src/main/kotlin/csh/back/domain/vote/user/entity/VoteUser.kt
package csh.back.domain.vote.user.entity

import csh.back.domain.trip.member.entity.TripMember
import csh.back.domain.vote.item.entity.VoteItem
import csh.back.domain.vote.vote.entity.Vote
import csh.back.global.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint

@Entity
@Table(
    name = "trip_vote_users",
    uniqueConstraints = [
        UniqueConstraint(name = "uk_vote_user", columnNames = ["trip_vote_id", "trip_member_id"]),
    ],
)
class VoteUser(
    @field:ManyToOne(fetch = FetchType.LAZY)
    @field:JoinColumn(name = "trip_vote_id", nullable = false)
    val vote: Vote,
    voteItem: VoteItem,
    @field:ManyToOne(fetch = FetchType.LAZY)
    @field:JoinColumn(name = "trip_member_id", nullable = false)
    val tripMember: TripMember,
    updateCount: Int,
) : BaseEntity() {

    @field:ManyToOne(fetch = FetchType.LAZY)
    @field:JoinColumn(name = "trip_vote_item_id", nullable = false)
    var voteItem: VoteItem = voteItem
        protected set

    @field:Column(nullable = false)
    var updateCount: Int = updateCount
        protected set

    fun updateVoteItemAndincreaseUpdateCount(voteItem: VoteItem) {
        this.voteItem = voteItem
        this.updateCount++
    }
}